src/domain/hooks/hooks.repository.ts

```diff
@@ -328,6 +348,10 @@ export class HooksRepositoryWithNotifications implements IHooksRepository {
         );
         break;
       case ConfigEventType.CHAIN_UPDATE:
+        // As the chains have been updated, we need to clear the memoized function cache.
+        if (this.isSupportedChainMemo.cache.clear) {
```

Comment:
Suggestion: What do you think about creating a private method to handle this condition?
The current method is nearly 300 lines long, so splitting it could improve readability. It’s fine for now, but it seems like the entire method could benefit from being refactored.

Reply:
Absolutely! Good point, I think we could move all the code within each `case` to private methods to improve the overall readability. Are you OK if I do that on a separate PR?

Reply:
@hectorgomezv Absolutely, that's a good idea.

Reply:
I've created an issue to keep track of it 👍 

https://github.com/safe-global/safe-client-gateway/issues/1926